### ISSUE TITLE
Add courses and people names to JSON API

### ISSUE THREAD

**Issue by jirutka** [cvut/sirius#96]
Fittable needs to display full names of people and (localized) names of courses. This information is already present in iCalendar representation, but not in JSON.

Sending another request for each course/person is unacceptable (performance), so we must figure out some way how to embed this information into JSON representation of the events collection. I see two possibilities:

1. Add fields to the event’s object (warn: event may have hundreds of people!).
2. Include people and courses in the events representation as [“compound documents”](http://jsonapi.org/format/#document-compound-documents) as defined in JSON API.

The second method seems better for me. What do you think @jnv and @flexik?

**Comment by jnv** [comment]
Courses names: Should there be both names in Czech and English, or should this be decided by URL parameter and/or HTTP header?

**Comment by jnv** [comment]
As for users' names, I agree that compounded resource makes more sense. Since we are far behind the current JSON-API spec to the point we have completely diverted, I propose the following:

```javascript
// GET /events?include=people

{
  events: [...]
  people: [
    {
      id: "jirutjak"
      name: "Ing. Jakub Jirůtka"
    },
    ...
  ]
}
```

**Comment by jnv** [comment]
As for courses names, there's no reason to duplicate these with each event, thus this will be also compounded.

```javascript
// GET /events?include=courses
// Accept-Language: cs
{
  events: [...]
  courses: [
    {
      id: "MI-RUB",
      name: "Programování v Ruby"
    }
  ]
}
```

We need to polish the locale negotiation though. If client requires multiple locales, it will be stored under separate attribute (e.g. object `names`). But I think the default behaviour should always return a single locale (i.e. Czech), even when no preference is specified.

**Comment by jirutka** [comment]
One of you already decided how to represent multilang strings (snippet from real data):

```json
{
  "events": {
    "id": 65630,
    "name": {
      "cs": "PJP Odevzdávání semestrálek"
    },
    "note": {
      "cs": "A1256"
    },
    "sequence_number": 14,
    "starts_at": "2015-06-16T15:00:00.000+02:00",
    ...
  }
}
```

So we have a problem…

**Comment by jnv** [comment]
Oh, yeah I think we are just passing the hstore hash for events name. Well,
screw it, let's just return both languages for courses too. We can review
this in v2.
On Jun 16, 2015 8:00 PM, "Jakub Jirutka" <notifications@github.com> wrote:

One of you already decided how to represent multilang strings (snippet from
real data):

{
  "events": {
    "id": 65630,
    "name": {
      "cs": "PJP Odevzdávání semestrálek"
    },
    "note": {
      "cs": "A1256"
    },
    "sequence_number": 14,
    "starts_at": "2015-06-16T15:00:00.000+02:00",
    ...
  }
}

So we have a problem…

—
Reply to this email directly or view it on GitHub
<https://github.com/cvut/sirius/issues/96#issuecomment-112515141>.

**Comment by jnv** [comment]
Alright, we've been thinking and instead of providing `people` as a single monolithic compounded resource, we should instead split this up into `students` and `teachers`.

Three reasons:

- You won't need students' names 99 % of time
- For privacy reasons you may want to prevent access to students' names
- We are linking to `students` and `teachers` separately anyway.

**Comment by jnv** [comment]
This is mostly implemented in #102; the outstanding stuff are singular resources, which is a lower priority for now.

**Comment by jirutka** [comment]
Great, thanks! :+1: :smile_cat:

I’ve deployed it to staging (with temporary search-api merged).
